spec/models/category_rspec.rb

```diff
@@ -3,4 +3,36 @@
 describe Category do
   it {should have_many(:videos)}
   it {should validate_uniqueness_of(:name)}
+
+  describe "#recent_videos" do
+    it "should return an empty array if no videos found" do
+      category = Category.create(name: "syfy")
+      expect(category.recent_videos).to eq([])
+    end
+    it "should return an array of one if only found one video" do
+      category = Category.create(name: "syfy")
+      stargate = Video.create(title: "stargate", description: "stargate video", category: Category.find_by(name: "syfy"))
+      expect(category.recent_videos).to eq([stargate])
+    end
+    it "should return 6 videos at most" do
+      category = Category.create(name: "syfy")
+      stargate_seasons = 10
+      stargate_seasons.times do |season|
+        stargate = Video.create(title: "stargate season #{season + 1}", description: "video from season #{season + 1}", 
+          category: Category.find_by(name: "syfy"))
+      end
+      expect(category.recent_videos.count).to eq(6)
+    end
+    it "should be ordered by created_at ascending(last video first)" do
+      category = Category.create(name: "syfy")
+      stargate_seasons = 10
+      seasons_array = []
+      stargate_seasons.times do |season|
+        stargate = Video.create(title: "stargate season #{season + 1}", description: "video from season #{season + 1}", 
+          category: Category.find_by(name: "syfy"))
+        seasons_array << stargate
+      end

```

Comment:
In general you want your specs to be the most readable as possible - this is not a good way. I would recommend having each item created separately and also all you need is 7 items right. 10 is overkill. Whatever works for 7 should work for 10 in the same sense.